gstreamer/src/message.rs

```diff
@@ -1184,12 +1184,13 @@ macro_rules! message_builder_generic_impl {
                 }
 
                 {
-                    let s = StructureRef::from_glib_borrow_mut(
-                        ffi::gst_message_get_structure(msg) as *mut _
-                    );
+                    let structure = ffi::gst_message_get_structure(msg);
+                    if !structure.is_null() {
+                        let structure = StructureRef::from_glib_borrow_mut(structure as *mut _);
```

Comment:
This means that extra fields are now silently ignored. It's the best we can do without breaking API now, but for the next API breakage the other fields API should just be removed for messages. Doing it properly requires additions to GStreamer (similar to what exists for events/queries).

I'll merge your patch to the 0.10 branch (once the question above is solved) and make a new release soon, and remove the API in master.

Reply:
Yes indeed... I can do the change in GStreamer if you want me to.

Reply:
That would be great :) let's try to get that in before 1.14

Reply:
All right!

Reply:
Just a comment: I don't think it is an issue using `ffi::gst_message_get_structure` in the `build` method. There is no extra fields at this point. I think it is better to use `ffi::gst_message_get_structure` and `ffi::gst_event_get_structure` in the `build`s as they don't allocate `structure`s when they are not needed. We should add `get_writable_structure` for both `gst::Message` and `gst::Event` for the user to add extra fields after the build.

Reply:
They can only be safely added during the build. Otherwise you need to ensure that the refcount is 1 for the message to be writable, etc

I'd prefer to have messages and events immutable after creation. We can prevent the allocation by not calling anything if no other fields

Reply:
I see! Let me a couple of minutes to do that too.

Reply:
BTW: https://bugzilla.gnome.org/show_bug.cgi?id=792928

Reply:
Thanks, merged that one